geniusyield-market-maker/src/GeniusYield/MarketMaker/MakerBot.hs

```diff
@@ -115,19 +130,129 @@ buildAndSubmitActions user@User {uColl, uStakeCred} providers netId ua di = flip
     gyAwaitTxConfirmed providers awaitTxParams tid
     logInfo $ printf "Tx successfully seen on chain with %d confirmation(s)" numConfirms
 
+
+-----------------------------------------------------------------------
+-------------------------- STATE MACHINE ------------------------------
+
+mbStateMachine ∷ Strategy
+               → MakerBot
+               → GYNetworkId
+               → GYProviders
+               → PricesProviders
+               → DEXInfo
+               → MBFret
+               → StateT MBFret IO ()
+mbStateMachine runStrategy mb@MakerBot {mbUser, mbDelay, mbToken} netId providers pp di mbFret = do
+  let cfg          = ppaCommonCfg . pricesAggregatorPP $ pp
+      respiteDelay = ceiling $ pccRespiteDelayFactor cfg * fromIntegral mbDelay
+
+  case mbFret of
+    MBReady → do
+      (newActions, controller) ← lift $ runStrategy pp mbUser mbToken
+      case controller of
+        UACNormal     → do
+          lift $ buildAndSubmitActions mbUser providers netId newActions di
+
+          lift $ gyLogInfo providers logNS "---------- Done for the block! -----------"
+          lift $ threadDelay mbDelay
+
+        UACSpooked1   → do
+          put MBSpooked1 { mbs1Relax = 0, mbs1Worse = 0 }
+          lift $ gyLogInfo providers logNS "Closing all orders due to price mismatch among Prices Providers"
+
+          lift $ cancelAllOrders' mb netId providers di
+          lift $ threadDelay respiteDelay
+
+        UACSpooked2 e → do
+          put MBSpooked2 { mbs2Relax = 0 }
+          lift $ gyLogWarning providers logNS $ "Closing all orders due to: " ++ e
+
+          lift $ cancelAllOrders' mb netId providers di
+          lift $ threadDelay respiteDelay
+
+    MBSpooked1 {..} → do
+      if mbs1Relax > pccAfterExitRelaxAim1 cfg then do
```

Comment:
Maybe it should be `>=`, also need to check for other places where such inequality test is performed.

Reply:
Indeed; done.